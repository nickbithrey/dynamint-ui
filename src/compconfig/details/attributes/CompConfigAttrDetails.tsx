import * as React from 'react';
import { Form, IForm } from 'lib/form';
import { Field } from 'lib/form/field';
import { ICompConfigAttribute } from 'model/compconfigattr';

export interface IDetails {
    id: string | number;
    loading: boolean;
    item: ICompConfigAttribute;
    update: (attribute: ICompConfigAttribute) => void;
    back: () => void;
}

const Details = ({loading, item, update, back}: IDetails) => {
    return (
        <Form entity={item} storageKey={'compConfigAttributeDetails'} save={update} back={back}>
            <Field
                type={'text'}
                name={'name'}
                label={'Name'}
                value={item.name}
                update={() => {}}
            />
            <Field
                type={'text'}
                name={'type'}
                label={'Type'}
                value={item.type}
                update={() => {}}
            />
            <Field
                type={'text'}
                name={'defaultValue'}
                label={'Default Value'}
                value={item.defaultValue}
                update={() => {}}
            />
        </Form>
    );
}

export default Details;