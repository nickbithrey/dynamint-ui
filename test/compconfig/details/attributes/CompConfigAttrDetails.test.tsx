import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import CompConfigAttrDetails, { IDetails } from '~/compconfig/details/attributes/CompConfigAttrDetails';

describe('<CompConfigAttrDetails />', () => {
    
    const props: IDetails = {
        id: 'id',
        item: {
            name: 'val1',
            type: 'type',
            defaultValue: 'default'
        },
        update: jest.fn(),
        back: jest.fn()
    };

    let wrapper: ShallowWrapper;

    beforeEach(() => {
        wrapper = shallow(<CompConfigAttrDetails {...props} />);
    });
    
    it('renders a valid tag and matches snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
});