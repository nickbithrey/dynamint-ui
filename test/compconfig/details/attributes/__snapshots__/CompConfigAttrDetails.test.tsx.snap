// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompConfigAttrDetails /> renders a valid tag and matches snapshot 1`] = `
<Form
  back={[MockFunction]}
  entity={
    Object {
      "defaultValue": "default",
      "name": "val1",
      "type": "type",
    }
  }
  save={[MockFunction]}
  storageKey="compConfigAttributeDetails"
>
  <Component
    label="Name"
    name="name"
    type="text"
    update={[Function]}
    value="val1"
  />
  <Component
    label="Type"
    name="type"
    type="text"
    update={[Function]}
    value="type"
  />
  <Component
    label="Default Value"
    name="defaultValue"
    type="text"
    update={[Function]}
    value="default"
  />
</Form>
`;
